tests/resource/test_specs.py

```diff
@@ -366,3 +367,110 @@ def circuit():
         # At the device level, approximations don't exist anymore and therefore
         # we should expect an empty errors dictionary.
         assert dev_specs["errors"] == {}
+
+
+class TestSpecsGraphModeExclusive:
+    """Tests for qml.specs features that require graph mode enabled.
+    The legacy decomposition mode should not be able to run these tests.
+
+    NOTE: All tests in this suite will auto-enable graph mode via fixture.
+    """
+
+    @pytest.fixture(autouse=True)
+    def enable_graph_mode_only(self):
+        """Auto-enable graph mode for all tests in this class."""
+        qml.decomposition.enable_graph()
+        yield
+        qml.decomposition.disable_graph()
```

Comment:
I believe there is already a fixture that enables graph mode, can you check in the conftest? Just apply that fixture on the test class.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/8265/commits/49d7428669beee9a81bdcae863e050baca90f153